backend/src/main/kotlin/org/darkSolace/muse/userModule/service/UserTagService.kt

```diff
@@ -0,0 +1,46 @@
+package org.darkSolace.muse.userModule.service
+
+import org.darkSolace.muse.userModule.model.User
+import org.darkSolace.muse.userModule.model.UserTags
+import org.darkSolace.muse.userModule.repository.UserRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class UserTagService(@Autowired val userRepository: UserRepository) {
+    @Transactional
+    fun addTagToUser(user: User, tag: UserTags): User {
+        val changedUser =
+            if (user.id == null) {
+                userRepository.findByUsername(user.username)
+            } else {
+                user
+            }
+
+        changedUser.userTags.add(tag)
+        if (tag.name.endsWith("_INACTIVE")) {
+            val activeTag = UserTags.values().first { it.name == tag.name.removeSuffix("_INACTIVE") }
+            changedUser.userTags.remove(activeTag)
```

Comment:
will this not throw an error if the active tag doesn't exist?

Reply:
It would have thrown an exception - changed the behavior.